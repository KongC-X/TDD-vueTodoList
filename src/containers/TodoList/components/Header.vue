<template>
    <div class="header">
      <div class="header-content">
        TodoList
        <input
          class="header-input"
          data-test="input"
          v-model="inputValue"
          @keyup.enter="addTodoItem"
          placeholder="Add TodoItem"
        />
      </div>
    </div>
</template>

<script>
export default {
  name: 'Header',
  data () {
    return {
      inputValue: ''
    }
  },
  methods: {
    addTodoItem () {
      if (this.inputValue) {
        this.$emit('add', this.inputValue)
        this.inputValue = ''
      }
    }
  }
}
</script>

<style lang="stylus" scoped>
.header{
  line-height 60px
  background #333
}
.header-content{
  width 600px
  margin 0 auto
  color #ffffff
  font-size 28px
}
.header-input{
  float right
  width 360px
  margin-top 16px
  line-height 24px
  border-radius 5px
  outline none
  color #333
  text-indent: 2em
  text-indent 10px
}
</style>
