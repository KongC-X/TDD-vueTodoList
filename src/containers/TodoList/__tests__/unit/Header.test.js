import { shallowMount } from '@vue/test-utils'
import Header from '../../components/Header'
import { findTestWrapper } from '../../../../utils/testUtils'

describe('Headerコンポーネント', () => {
  it('style変化時にお知らせ', () => {
    const wrapper = shallowMount(Header)
    expect(wrapper).toMatchSnapshot()
  })

  it('input存在します', () => {
    const wrapper = shallowMount(Header)
    const input = findTestWrapper(wrapper, 'input')
    expect(input.exists()).toBe(true)
  })

  it('inputの初期値はnull', () => {
    const wrapper = shallowMount(Header)
    const inputValue = wrapper.vm.$data.inputValue
    expect(inputValue).toBe('')
  })
  it('inputのvalueが変化すれば、データも共に変化する', () => {
    const wrapper = shallowMount(Header)
    const input = findTestWrapper(wrapper, 'input')
    input.setValue('dell lee')
    const inputValue = wrapper.vm.$data.inputValue
    expect(inputValue).toBe('dell lee')
  })
  it('inputのvalueがnullの場合enter,無反応', () => {
    const wrapper = shallowMount(Header)
    const input = findTestWrapper(wrapper, 'input')
    input.setValue('')
    input.trigger('keyup.enter')
    expect(wrapper.emitted().add).toBeFalsy()
  })
  it('inputのvalueが存在すれば、keyup.enter,$emitを実行,inputのvalueがnullになる', () => {
    const wrapper = shallowMount(Header)
    const input = findTestWrapper(wrapper, 'input')
    input.setValue('dell lee')
    input.trigger('keyup.enter')
    expect(wrapper.emitted().add).toBeTruthy()
    const inputValue = wrapper.vm.$data.inputValue
    expect(inputValue).toBe('')
  })
})
