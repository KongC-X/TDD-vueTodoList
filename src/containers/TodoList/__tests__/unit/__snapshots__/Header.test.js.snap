// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headerコンポーネント style変化時にお知らせ 1`] = `
<div class="header">
  <div class="header-content">
    TodoList
    <input data-test="input" placeholder="TodoItem" class="header-input"></div>
</div>
`;
